### PR TITLE
test(mobile): cover getDocumentsPath on web; document manual Chrome run

### LINKED ISSUE #3076: test(library): add runtime web coverage for getDocumentsPath
## Context

Follow-up from PR #3029: https://github.com/divinevideo/divine-mobile/pull/3029

PR #3029 adds kIsWeb handling in getDocumentsPath() so Library drafts/clips do not call path_provider on web. The current unit test covers resolvePath(..., ""), and the Library regression test verifies that the plugin exception text does not render.

## Problem

The test comment says the web branch is exercised by flutter build web, but a successful web build is compile-time coverage, not runtime coverage of getDocumentsPath().

## Suggested fix

Either:

- Add a Chrome-platform test that directly asserts await getDocumentsPath() returns an empty string on web, or
- Reword/remove the test comment and document runtime verification elsewhere if Chrome tests are not practical in CI.

## Acceptance criteria

- The test/documentation no longer implies that flutter build web proves runtime behavior.
- If practical, getDocumentsPath() has direct web runtime coverage under a Chrome-platform test.
- Existing Library regression coverage still passes.

### PR DESCRIPTION
**Pull request by evgenibir**
## Description

Follow-up to the web-safe `getDocumentsPath()` work (#3029 / #2994).

**Problem:** VM `flutter test` shards skip assertions guarded with `kIsWeb`, so CI never executes a runtime check that `getDocumentsPath()` returns `''` on web. Earlier wording also suggested that `flutter build web` was enough; it only validates compilation, not that branch at runtime.

**What this branch does (net vs `main`):**

- Adds a **`getDocumentsPath`** test in `mobile/test/utils/path_resolver_test.dart` that asserts an empty string on web, with **`skip` when `!kIsWeb`** so normal CI keeps using VM-only shards.
- Replaces misleading comments: **`flutter build web` ≠ proof of the web runtime path**; documents that the web assertion is exercised with  
  `cd mobile && flutter test test/utils/path_resolver_test.dart --platform chrome` (manual, not part of Mobile CI).
- Extends **`CONTRIBUTING.md`** with the same manual Chrome expectation when changing `getDocumentsPath` / web document paths (including **`CHROME_EXECUTABLE`** / **`xvfb-run -a`** on headless Linux where needed).
- Updates **`mobile/lib/utils/path_resolver.dart`** dartdoc to point at the test and the manual Chrome command.
- Adjusts **`mobile/test/flutter_test_config.dart`** so that on **`kIsWeb`** it skips **`loadAppFonts()`** and **Alchemist** and runs **`testMain()`** directly—otherwise headless Chrome can sit on `loading …` with almost no CPU during global golden setup.

**Verification:** From `mobile/`, run `flutter test test/utils/path_resolver_test.dart` on the VM (expect the web test skipped) and again with `--platform chrome` (expect it not skipped and green).

**Related Issue:** Relates to #3076

## Type of Change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code refactor
- [ ] Build configuration change
- [x] Documentation
- [ ] Chore

### CODE REVIEW

File: mobile/test/utils/path_resolver_test.dart [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
 
+import 'package:flutter/foundation.dart' show kIsWeb;
 import 'package:flutter_test/flutter_test.dart';
 import 'package:openvine/utils/path_resolver.dart';
 
 void main() {
+  group('getDocumentsPath', () {
+    test(
+      'returns empty string on web without using path_provider',
+      () async {
```
Comment (NotThatKindOfDrLiz) [inline mobile/test/utils/path_resolver_test.dart:16]: This is the right coverage shape for the web branch: it exercises getDocumentsPath() in an actual web runtime instead of relying on a compile-only web build.

File: mobile/test/flutter_test_config.dart [hunk 2]
```diff
@@ -4,13 +4,20 @@
...
 import 'package:golden_toolkit/golden_toolkit.dart';
 import 'test_setup.dart';
 
 Future<void> testExecutable(FutureOr<void> Function() testMain) async {
   // Set up test environment with plugin mocks (secure_storage, path_provider, etc.)
   setupTestEnvironment();
 
+  // Web / `flutter test --platform chrome`: skip golden font loading and Alchemist.
+  // Those paths can stall headless Chrome with almost no CPU while `loading …` is shown.
+  if (kIsWeb) {
```
Comment (NotThatKindOfDrLiz) [inline mobile/test/flutter_test_config.dart:17]: This kIsWeb guard is appropriately scoped. It avoids the Chrome runner hang without changing the normal VM shard setup.

**Review by NotThatKindOfDrLiz — APPROVED** [review]
Approved. This is a clean follow-up for the web coverage gap from #3029 / #3076.

The new test exercises getDocumentsPath() in an actual web runtime, and the skip message plus CONTRIBUTING note make the manual Chrome run explicit. The flutter_test_config.dart guard is scoped to kIsWeb, so normal VM test behavior is unchanged.

CI is green. Ready to merge.
